apps/src/JSInterpreter.js

```diff
@@ -815,39 +870,42 @@ JSInterpreter.prototype.shouldBlockCustomMarshalling_ = function (name, obj,
  *
  * Fetch a property value from a data object.
  * @param {!Object} interpeter Interpreter instance.
- * @param {!Function} baseGetProperty Original getProperty() implementation.
  * @param {!Object} obj Data object.
  * @param {*} name Name of property.
  * @return {!Object} Property value (may be UNDEFINED).
  */
 JSInterpreter.prototype.getProperty = function (
     interpreter,
-    baseGetProperty,
     obj,
     name) {
   name = name.toString();
   var nativeParent;
-  if (obj.isCustomMarshal ||
-      (obj === this.globalScope &&
-          (!!(nativeParent = this.customMarshalGlobalProperties[name])))) {
-    if (this.shouldBlockCustomMarshalling_(name, obj, nativeParent)) {
+  var customMarshalValue;
+  if (obj.isCustomMarshal) {
+    if (this.shouldBlockCustomMarshalling_(name, obj)) {
       return interpreter.UNDEFINED;
-    }
-    var value;
-    if (obj.isCustomMarshal) {
-      value = obj.data[name];
     } else {
-      value = nativeParent[name];
+      customMarshalValue = obj.data[name];
     }
-    var type = typeof value;
-    if (type === 'number' || type === 'boolean' || type === 'string' ||
-        type === 'undefined' || value === null) {
-      return interpreter.createPrimitive(value);
+  } else {
+    var hasProperty = JSInterpreter.baseHasProperty.call(interpreter, obj, name);
+    if (!hasProperty &&
+        obj === this.globalScope &&
+        !!(nativeParent = this.customMarshalGlobalProperties[name]) &&
```

Comment:
I would prefer not to do assignment inside of conditionals. I'm a little surprised our linter doesn't complain about it.